src/visip/dev/dtype_new.py

```diff
@@ -0,0 +1,582 @@
+        t, var_map, new_map = to_typing_map(type.arg, var_map, new_map)
+        return dtype.Constant[t], var_map, new_map
+
+
+    # Class
+    if isinstance(type, Class):
+        return type.origin_type, var_map, new_map
+
+    # Enum
+    if isinstance(type, Enum):
+        return type.origin_type, var_map, new_map
+
+
+    # Any
+    if isinstance(type, Any):
+        return typing.Any, var_map, new_map
+
+    # NoneType
+    if isinstance(type, NoneType):
+        return builtins.type(None), var_map, new_map
+
+
+    raise TypeError("Not supported type.")
+
+
+def is_equaltype(type, other):
+    # Const
+    if isinstance(type, Const):
+        if isinstance(other, Const):
+            return is_equaltype(type.arg, other.arg)
+        elif isinstance(type.arg, Union):
+            return is_equaltype(Union([type]), other)
+        elif isinstance(other, Union):
+            return is_equaltype(other, type)
+
+    # Any
+    elif isinstance(type, Any):
+        return other is type
+
+    # TypeVar
+    elif isinstance(type, TypeVar):
+        return type is other
+
+    # Union
+    elif isinstance(type, Union):
+        if len(type.args) == 1:
+            return is_equaltype(type.args[0], other)
+        elif isinstance(other, Union) and len(type.args) == len(other.args):
+            for arg in type.args:
+                eq = False
+                for oarg in other.args:
+                    if is_equaltype(arg, oarg):
+                        eq = True
+                        break
+                if not eq:
+                    return False
+            return True
+
+    elif isinstance(other, Union):
+        return is_equaltype(other, type)
+
+    # NewType
+    elif isinstance(type, NewType):
+        if other is type:
+            return True
+
+    # Int, Float, Bool, Str, NoneType
+    elif isinstance(type, (Int, Float, Bool, Str, NoneType)):
+        return other is type
+
+    # Class
+    elif isinstance(type, Class):
+        return isinstance(other, Class) and type.origin_type is other.origin_type
+
+    # Enum
+    elif isinstance(type, Enum):
+        return isinstance(other, Enum) and type.origin_type is other.origin_type
+
+    # Tuple
+    elif isinstance(type, Tuple):
+        if isinstance(other, Tuple) and len(type.args) == len(other.args):
+            for arg, oarg in zip(type.args, other.args):
+                if not is_equaltype(arg, oarg):
+                    return False
+            return True
+
+    # List
+    elif isinstance(type, List):
+        if isinstance(other, List):
+            return is_equaltype(type.arg, other.arg)
+
+    # Dict
+    elif isinstance(type, Dict):
+        if isinstance(other, Dict):
+            return is_equaltype(type.key, other.key) and is_equaltype(type.value, other.value)
+
+    return False
+
+
+def is_subtype(type, typeinfo):
+    b, _ = is_subtype_map(type, typeinfo, {})
+    return b
+
+
+def is_subtype_map(type, typeinfo, var_map, const=False):
```

Comment:
Const means that only Value can be connected at the input. Therefore we need probably no full support for subtype 
to Const types.